sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/collectionOperations.scala

```diff
@@ -902,33 +902,45 @@ case class SortArray(base: Expression, ascendingOrder: Expression)
 
 
 /**
- * Sorts the input array in ascending order according to the natural ordering of
+ * Sorts the input array in ascending / descending order according to the natural ordering of
  * the array elements and returns it.
  */
-// scalastyle:off line.size.limit
+// scalastyle:on line.size.limit
 @ExpressionDescription(
   usage = """
     _FUNC_(array) - Sorts the input array in ascending order. The elements of the input array must
-      be orderable. Null elements will be placed at the end of the returned array.
+      be orderable. Null elements will be placed at the beginning of the returned array in ascending
+       order or at the end of the returned array in descending order.
   """,
   examples = """
     Examples:
       > SELECT _FUNC_(array('b', 'd', null, 'c', 'a'));
-       ["a","b","c","d",null]
+       [null,"a","b","c","d"]
```

Comment:
It looks like you changed behavior here, and I don't think we can do that.

Reply:
I don't think we can change the behavior either without adding a legacy config and a migration guide.

Btw, if we unify the behaviors, then we don't need to keep the two implementations?
We might want to deprecate one of the two.

Reply:
I agree, I kept the original behaviour of array_sort. 
Null will be still placed at the end of the array in ascending order and also in descending.

I completely agree about deprecating sort_array, to simplify.